x-pack/plugins/security_solution/public/management/cypress/tasks/load_endpoint_data.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import { BASE_ENDPOINT_ROUTE } from '../../../../common/endpoint/constants';
+import { runEndpointLoaderScript } from './run_endpoint_loader';
+
+// Checks for Endpoint data and creates it if needed
+export const loadEndpointDataForEventFiltersIfNeeded = () => {
+  cy.request({
+    method: 'POST',
+    url: `${BASE_ENDPOINT_ROUTE}/suggestions/eventFilters`,
+    body: {
+      field: 'agent.type',
+      query: '',
+    },
```

Comment:
Kind of learning more about cypress tests through these prs, but how do you know when you need a body for the request and when you don't?

Reply:
Thanks for asking @parkiino. I usually just make the API call locally using curl/postman before adding it to the test suite.

Reply:
that's a great idea, @ashokaditya, thanks!

I saw you had a similar implementation, so I also renamed the helper files I similarly created, to minimise the merge issues we're gonna have (in this commit https://github.com/elastic/kibana/pull/149324/commits/8eae2fa9cf40a33b1324cf706004f17021efa6ca)

regarding the `header` and `body`: I haven't checked it before, just extracted this helper function that was written before. now I checked it, and the `POST api/endpoint/suggestions/eventFilters` endpoint really needs the header and the body - in opposite of the `GET /api/endpoint/metadata` endpoint you were using.

so the two functions are very similar, but they have different goals: `loadEndpointDataForEventFiltersIfNeeded()` wants the event filter suggestions, `loadEndpointIfNoneExist()` wants the endpoints - different condition, same action. I think we should keep both, in case later we differentiate the actions (e.g. the endpoint loader script parameters). what do you think?

Reply:
Oh yes quite right about the `/suggestions/eventFilters` API needed the `kbn-xsrf` and `body` 👍 Sorry I missed that earlier.

Reply:
Yeah, let's keep both functions. I like that they are abstracted based on what context they need to run in. Easier to read and understand when you're reading the test files only.